src/modules/aws_ec2/mappers/instance.ts

```diff
@@ -290,6 +338,85 @@ export class InstanceMapper extends MapperBase<Instance> {
     _res => undefined,
   );
 
+  // given an instance reads the mapping from the associate AMI , match
+  // it with the current mapped volumes and generate the final mapping
+  async generateBlockDeviceMapping(
+    ctx: Context,
+    ami: string,
+    maps: InstanceBlockDeviceMapping[],
+    instance: Instance,
+    encrypted: boolean,
+  ) {
+    // start reading the block device mapping from the image
+    const client = await ctx.getAwsClient(instance.region);
+    const amiImage = (await this.describeImages(client.ec2client, [ami]))?.Images?.pop();
+
+    if (amiImage) {
+      const imageMapping = amiImage.BlockDeviceMappings;
+
+      // check if there is any mapped volume that doesn't exist on instance mapping, and error
+      // or if there is any volume mapped that is not set as root
```

Comment:
it means that i look at is_root_device_flag. If there is an instance_block_device_mapping associated to this instance but is not set as root device it will throw an error, as only the volumes flagged as root device can be used on instance creation.
The alternative if we find a case like that (an instance that is not created, but with some mapping already associated), is to ignore this volume and later we can associate as an extra one.